### PR TITLE
Closes #3226: Adds parameterization to test_shuffle and test_permutation

### LINKED ISSUE #3226: Update `random_test`

### PR DESCRIPTION
**Pull request by drculhane**
This includes the same parameterizations to test_shuffle and test_permutation (in PROTO-tests/tests/random_test.py) as a couple of weeks ago, updated to the newest master.

Closes #3226

### CODE REVIEW

**Review by ajpotts — APPROVED** [review]
Looks good.  Thanks!

**Review by jaketrookman — APPROVED** [review]

**Comment by stress-tess** [comment]
Just making sure we capture this exchange from the previous iteration of this PR:
https://github.com/Bears-R-Us/arkouda/pull/3230#discussion_r1651567928

File: PROTO_tests/tests/random_test.py [hunk 1]
```diff
@@ -54,62 +56,70 @@ def test_integers(self):
...
+        # ints are checked for equality; floats are checked for closeness
 
-        float_pda = rng.uniform(-(2**32), 2**32, 10)
-        pda_copy = float_pda[:]
-        rng.shuffle(float_pda)
-        # verify all the same elements are in permutation as the original
-        assert (ak.sort(float_pda) == ak.sort(pda_copy)).all()
+        check = lambda a, b, t: (
+            (a == b).all() if t is ak.int64 else np.allclose(a.to_list(), b.to_list())
+        )
```
Comment (stress-tess) [inline PROTO_tests/tests/random_test.py:66]: I like this a lot! super nifty

**Review by stress-tess — APPROVED** [review]
looks great! sorry for dragging my feet on reviewing this one 😅

**Comment by stress-tess** [comment]
also just a reminder, it helps to use closing keywords to link the PR and issue automatically. Also including the issue number in the PR title is super helpful when writing release notes

**Pull request merged**